File: 2022-2023/sourcecode/lab/lab9/RoomRelationsJava/app/src/main/java/pl/edu/uwr/pum/roomrelationsjava/model/entities/relations/FacultyWithStudents.java
package pl.edu.uwr.pum.roomrelationsjava.model.entities.relations;

import androidx.room.Embedded;
import androidx.room.Relation;

import java.util.List;

import pl.edu.uwr.pum.roomrelationsjava.model.entities.Faculty;
import pl.edu.uwr.pum.roomrelationsjava.model.entities.Student;

public class FacultyWithStudents {
    @Embedded
    public Faculty faculty;

    @Relation(
            parentColumn = "facultyName",
            entityColumn = "facultyName"
    )

    public List<Student> studentList;
}

File: 2022-2023/sourcecode/lab/lab9/RoomRelationsKotlin/app/src/main/java/pl/edu/uwr/pum/roomrelationskotlin/model/entity/relations/FacultyWithStudents.kt
package pl.edu.uwr.pum.roomrelationskotlin.model.entity.relations

import androidx.room.Embedded
import androidx.room.Relation
import pl.edu.uwr.pum.roomrelationskotlin.model.entity.Faculty
import pl.edu.uwr.pum.roomrelationskotlin.model.entity.Student

data class FacultyWithStudents (
    @Embedded val faculty: Faculty,

    @Relation(
        parentColumn = "facultyName",
        entityColumn = "facultyName"
    )

    val students: List<Student>
        )